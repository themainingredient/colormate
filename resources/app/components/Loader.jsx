import React from 'react';
import styled from 'styled-components';
import Globals, { flexCenter } from '../Global.styles';

const loader = require('../assets/washingTransparent.gif');

const { colors, fonts } = Globals;

const LoaderWrapper = styled.div`
  ${flexCenter};
  flex-direction: column;
  background-color: ${colors.TMIBlue};
  height: 100%;
  width: 100%;
  text-align: center;
`;

const TextBase = styled.h2`
  color: ${colors.White};
  font-family: ${fonts.Futura.bold};
`;

const LoaderTitle = styled(TextBase)`
  font-size: 20px;
`;

const LoaderContent = styled(TextBase)`
  font-size: 16px;
`;

const LoaderImage = styled.img`
  height: 195px;
  margin: 50px 0;
`;

const Loader = () => (
  <LoaderWrapper>
    <LoaderTitle>Scanning whole file...</LoaderTitle>
    <LoaderImage src={loader} alt='Loading...' />
    <LoaderContent>
      Sorting, cleaning and <br />
      separating your colors. <br />
      Damnnn, you messy{' '}
      <span role='img' aria-label='WinkyFace'>
        ðŸ˜‰
      </span>
    </LoaderContent>
  </LoaderWrapper>
);

export default Loader;
