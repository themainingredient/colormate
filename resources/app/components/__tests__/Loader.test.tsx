import React from 'react';
import { render, cleanup } from 'react-testing-library';
import Loader from '../Loader';

jest.mock('../../assets/text1Loader.svg', () => 'text-one-loader');
jest.mock('../../assets/text2Loader.svg', () => 'text-two-loader');

afterEach(cleanup);

describe('Loader.jsx', () => {
  test('renders and displays correctly', () => {
    const { container } = render(<Loader />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
