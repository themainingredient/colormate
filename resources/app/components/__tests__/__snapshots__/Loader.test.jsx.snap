// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader.jsx renders and displays correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #4E41FF;
  height: 100%;
  width: 100%;
  text-align: center;
}

.c1 {
  height: 195px;
  margin: 50px 0;
}

<div
  class="c0"
>
  <text-one-loader />
  <img
    alt="Loading..."
    class="c1"
    src="test-file-stub"
  />
  <text-two-loader />
</div>
`;
